kyo-core/shared/src/main/scala/kyo/resources.scala

```diff
@@ -20,9 +20,13 @@ object Resources:
                 }
         }
 
+    def acquireRelease[T](acquire: => T < IOs)(release: T => Unit < IOs): T < Resources =
+        acquire.map { resource =>
+            ensure(release(resource)).andThen(resource)
+        }
+
     def acquire[T <: Closeable](resource: => T): T < Resources =
```

Comment:
Maybe we could remove this method? It feels too unsafe since making a resource should typically require `IOs`.

Reply:
What if we add an IOs to the type?? Or are you suggesting that it's too unsafe to interop with `Closeable`s?
```
def acquire[T <: Closeable](resource: => T < IOs): T < Resources =
```

Reply:
Yeah, I'm not sure relying on Java's unsafe API is ideal but I'm ok with it and adding `IOs`.